common/setup.py

```diff
@@ -29,9 +22,28 @@
     maintainer='Amundsen TSC',
     maintainer_email='amundsen-tsc@lists.lfai.foundation',
     packages=find_packages(exclude=['tests*']),
-    install_requires=requirements_common,
+    install_requires=[
+        # Packages in here should rarely be pinned. This is because these
+        # packages (at the specified version) are required for project
+        # consuming this library. By pinning to a specific version you are the
+        # number of projects that can consume this or forcing them to
+        # upgrade/downgrade any dependencies pinned here in their project.
+        #
+        # Generally packages listed here are pinned to a major version range.
+        #
+        # e.g.
+        # Python FooBar package for foobaring
+        # pyfoobar>=1.0, <2.0
+        #
+        # This will allow for any consuming projects to use this library as
+        # long as they have a version of pyfoobar equal to or greater than 1.x
+        # and less than 2.x installed.
+        'Flask>=1.0.2',
+        'attrs>=19.0.0',
+        'marshmallow>=3.0',
```

Comment:
As we discussed I really wouldn't want to get back to this situation. One of the most notorious things that happened and triggered dependency unification is marshmallow mismatch and it was brought up by community quite often. And the way it's defined in requirements-common.txt is pretty much ok (it's not hard pin but a fair range).

Right now it's perfectly in sync but it can easily become inconsistent. I would rather filter out marshmallow from requirements-common.txt which I agree is uglier solution but at least keeps things consistent across the board.

Reply:
Ah I understand. However I think the point I had made in the PR description was missed. This list in common/setup.py does *not* need to be in sync with `requirements-common.txt`. It is okay for the requirements listed here to be a looser set. When `amundsen-common` is used in search or frontend or etc, the looser requirements here will not matter. 

Since I suspect the dependencies in requirements-common.txt will be advanced upwards, the looser requirements here will not have an effect on which versions are used in services.

Reply:
In other words, the actual requirements listed here in `common/setup.py` are the library's own requirements which only matter when the library is used outside of amundsen, for example in Airflow etc.